src/main/java/database/DatabaseConnection.java

```diff
@@ -0,0 +1,47 @@
+package database;
+
+import com.arangodb.ArangoDB;
+import utils.ConfigReader;
+
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.util.Properties;
+
+/**
+ * A singleton class carrying a database instance
+ */
+public class DatabaseConnection {
+    private ArangoDB arangoDriver;
+    private DatabaseHandler dbHandler;
```

Comment:
Please remove DatabaseHandler from here. As we have agreed before, DatabaseConnection is used to only initiate connections to database and return a singleton instance that provide the whole application with the database drivers.